src/SpeechModule.ts

```diff
@@ -1,43 +1,169 @@
-
 export type Action = () => void
 
-export type Func<T,TResult> = (item: T) => TResult;
+export type Func<T, TResult> = (item: T) => TResult;
+
+export module Speech {
+    export interface ISpeechRecognizer {
+        locale: string;
+        isStreamingToService: boolean;
+        referenceGrammarId: string; // unique identifier to send to the speech implementation to bias SR to this scenario
 
-export class SpeechSynthesizer {
-    private static instance : SpeechSynthesis.ISpeechSynthesizer = null;
+        onIntermediateResult: Func<string, void>;
```

Comment:
Why aren't these Action<string> ?

I could also be talked in to a more JS-like way of specifying these types